### ISSUE TITLE
Integrate Stable Baselines and Alternative RL Libraries for Enhanced RL Capabilities

### ISSUE THREAD

**Issue by csmangum** [Dooders/AgentFarm#269]
### Current RL Implementation Analysis

After examining the codebase, AgentFarm currently uses a **custom PyTorch-based DQN implementation** with the following architecture:

**Core Components:**
- `BaseDQNModule` - Complete DQN implementation with experience replay, target networks, epsilon-greedy exploration
- `BaseQNetwork` - Neural network with layer normalization, dropout, and shared encoder support
- `SharedEncoder` - Common feature extraction across action modules
- Separate action modules: `MoveModule`, `AttackModule`, `GatherModule`, `ReproduceModule`, `ShareModule`, `SelectModule`

**Key Features:**
- Double Q-Learning with soft target updates
- Experience replay buffer with configurable size
- Gradient clipping and weight initialization
- State caching for performance optimization
- Comprehensive logging and database integration
- Pydantic-based configuration system

**Current Limitations:**
- Monolithic DQN implementation - difficult to experiment with other RL algorithms
- No support for advanced RL algorithms (PPO, SAC, A2C, etc.)
- Limited algorithm comparison capabilities
- Custom implementation requires maintenance overhead

### Proposed Enhancement

**Objective:** Integrate Stable Baselines3 and other RL libraries while maintaining backward compatibility with the current custom DQN implementation.

**Benefits:**
- Access to state-of-the-art RL algorithms (PPO, SAC, A2C, TD3, etc.)
- Standardized implementations with proven performance
- Easy algorithm comparison and benchmarking
- Reduced maintenance burden for core RL functionality
- Flexibility to use custom PyTorch models when needed

### Implementation Approach

**1. Abstract RL Interface**
Create a unified interface that can work with both custom implementations and library-based ones:

```python
class RLAlgorithm(ABC):
    """Abstract base class for RL algorithms."""
    
    @abstractmethod
    def select_action(self, state: torch.Tensor) -> int:
        """Select action given current state."""
        pass
    
    @abstractmethod
    def store_experience(self, state, action, reward, next_state, done):
        """Store experience in replay buffer."""
        pass
    
    @abstractmethod
    def train(self, batch_size: int) -> Optional[float]:
        """Train the algorithm on a batch of experiences."""
        pass
    
    @abstractmethod
    def get_state_dict(self) -> dict:
        """Get model state for saving."""
        pass
    
    @abstractmethod
    def load_state_dict(self, state_dict: dict):
        """Load model state."""
        pass
```

**2. Stable Baselines Integration**
- Add Stable Baselines3 as optional dependency
- Create wrapper classes for popular algorithms
- Maintain compatibility with current Gym/PettingZoo environments

**3. Configuration System Enhancement**
Extend current Pydantic config to support algorithm selection:

```python
class RLConfig(BaseDQNConfig):
    algorithm: str = Field(default="custom_dqn", 
                          description="RL algorithm to use")
    algorithm_kwargs: Dict[str, Any] = Field(default_factory=dict,
                                           description="Algorithm-specific parameters")
    
    @field_validator("algorithm")
    def validate_algorithm(cls, v):
        valid_algorithms = ["custom_dqn", "ppo", "sac", "a2c", "ddpg", "td3"]
        if v not in valid_algorithms:
            raise ValueError(f"Algorithm must be one of: {valid_algorithms}")
        return v
```

**4. Backward Compatibility**
- Keep existing `BaseDQNModule` as default
- Allow seamless switching between algorithms
- Maintain current API for action modules

### Dependencies to Add

```python
# Add to requirements.txt
stable-baselines3>=2.0.0
gymnasium>=0.28.0  # Modern Gym replacement
shimmy>=1.0.0  # Environment wrappers
```

### Migration Path

**Phase 1:** Add abstract interface and Stable Baselines wrappers
**Phase 2:** Update configuration system
**Phase 3:** Add algorithm comparison utilities
**Phase 4:** Documentation and examples

### Expected Outcomes

1. **Enhanced Performance:** Access to more sophisticated algorithms
2. **Research Flexibility:** Easy comparison of different RL approaches
3. **Maintenance Reduction:** Leverage battle-tested implementations
4. **Extensibility:** Simple addition of new algorithms
5. **Benchmarking:** Built-in support for algorithm performance comparison

### Files to Modify

- `farm/actions/base_dqn.py` - Add abstract interface
- `farm/actions/config.py` - Extend configuration system
- `requirements.txt` - Add new dependencies
- Create `farm/actions/rl_algorithms/` - New directory for algorithm implementations
- Update action modules to use new interface

### Testing Requirements

- Unit tests for new algorithm wrappers
- Integration tests with existing action modules
- Performance benchmarks comparing algorithms
- Backward compatibility tests

### Related Issues

This enhancement would complement existing benchmarking infrastructure in `/benchmarks/` and provide more comprehensive RL evaluation capabilities.
